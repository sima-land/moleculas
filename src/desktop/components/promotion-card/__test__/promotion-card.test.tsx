import { render } from '@testing-library/react';
import { PromotionCard } from '..';

describe('PromotionCard', () => {
  beforeEach(() => {
    jest.useFakeTimers();
    jest.setSystemTime(new Date('2021-01-01T09:00:00'));
  });

  it('should renders correctly', () => {
    const { container, queryAllByTestId } = render(
      <PromotionCard
        title='Hello, world!'
        subtitle='Foo, bar, baz...'
        imageSrc='https://www.images.com/123'
        dueDate={new Date()}
      />,
    );

    expect(container).toMatchSnapshot();
    expect(queryAllByTestId('promotion-card')).toHaveLength(1);
  });

  it('should handle "promotionType" prop', () => {
    const { container, queryAllByTestId } = render(
      <PromotionCard
        title='Hello, world!'
        subtitle='Foo, bar, baz...'
        imageSrc='https://www.images.com/123'
        dueDate={new Date()}
        promotionType='volume-discount'
        volumeDiscount={23}
      />,
    );

    expect(container).toMatchSnapshot();
    expect(queryAllByTestId('promotion-card:name')).toHaveLength(1);
    expect(queryAllByTestId('promotion-card:banner-discount')).toHaveLength(1);
  });

  it('should have placeholder component property', () => {
    const { container, queryAllByTestId } = render(<PromotionCard.Placeholder />);

    expect(container).toMatchSnapshot();
    expect(queryAllByTestId('promotion-card:placeholder')).toHaveLength(1);
  });
});
