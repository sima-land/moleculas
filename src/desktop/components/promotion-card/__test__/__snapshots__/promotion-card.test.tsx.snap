// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PromotionCard should handle "promotionType" prop 1`] = `
<div>
  <div
    class="root"
    data-testid="promotion-card"
  >
    <a
      class="link"
    >
      <div
        class="banner"
      >
        <img
          alt="Hello, world!"
          class="image"
          src="https://www.images.com/123"
        />
        <div
          class="banner-content"
        >
          <div
            class="discount"
            data-testid="promotion-card:banner-discount"
          >
            скидки до
            <div
              class="percent"
            >
              23
              %
            </div>
          </div>
        </div>
      </div>
      <div
        class="info"
      >
        <div
          class="header"
        >
          <h4
            class="title"
          >
            Hello, world!
          </h4>
          <p
            class="subtitle"
          >
            Foo, bar, baz...
          </p>
        </div>
        <div
          class="footer"
        >
          <div
            class="name"
            data-testid="promotion-card:name"
          >
            Скидка за объём
          </div>
          <time
            data-testid="promotion-card:due-date"
          >
            до 1 янв
          </time>
        </div>
      </div>
    </a>
  </div>
</div>
`;

exports[`PromotionCard should have placeholder component property 1`] = `
<div>
  <div
    data-testid="promotion-card:placeholder"
  >
    <div
      class="part banner"
    />
    <div
      class="part text"
    />
    <div
      class="part text"
    />
  </div>
</div>
`;

exports[`PromotionCard should renders correctly 1`] = `
<div>
  <div
    class="root"
    data-testid="promotion-card"
  >
    <a
      class="link"
    >
      <div
        class="banner"
      >
        <img
          alt="Hello, world!"
          class="image"
          src="https://www.images.com/123"
        />
      </div>
      <div
        class="info"
      >
        <div
          class="header"
        >
          <h4
            class="title"
          >
            Hello, world!
          </h4>
          <p
            class="subtitle"
          >
            Foo, bar, baz...
          </p>
        </div>
        <div
          class="footer"
        >
          <time
            data-testid="promotion-card:due-date"
          >
            до 1 янв
          </time>
        </div>
      </div>
    </a>
  </div>
</div>
`;
